spark/src/main/scala/org/apache/spark/sql/delta/OptimisticTransaction.scala

```diff
@@ -1108,6 +1108,9 @@ trait OptimisticTransactionImpl extends TransactionalWrite
       val readRowIdHighWatermark =
         RowId.extractHighWatermark(snapshot).getOrElse(RowId.MISSING_HIGH_WATER_MARK)
 
+      val allTags = RowTracking.addPreservedRowTrackingTagIfNotSet(
+        protocol, metadata, tags)
+
```

Comment:
Blindly tagging all commits as preserving row tracking is pretty brittle: there may be a case (now or introduced in the future) that doesn't preserve row tracking and there's virtually no chance that we detect it and correctly mark row tracking information as not preserved.

We could instead add tags to the most relevant code paths to explicitly mark that row tracking is preserved, at least in inserts (WriteIntoDelta, InsertOnlyMergeExecutor) so that most data ingestion operations are covered

We should in any case have a test somewhere that checks that the preserved row tracking tag is set to true for inserts.

Reply:
Okay, let's think more about it, but for now let's keep the Preserving tests in the commands that we preserved Row Tracking (that I forgot to add in the previous Delta PRs).